### PR TITLE
Converted docker example images to use dataset outside of container image

### LINKED ISSUE #3061: Convert docker container examples to use external data files instead of packing them inside
**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is. What kind of performance improvements would you like to see with this new API?

Currently we build example docker containers by packing datasets inside of container image. It has two disadvantages (1) huge image size (2) it is inflexible because doesn't allow to run example on a different dataset. It is possible to map a filesystem directory inside a container with `-v` option to `docker run` command. This way we can let user to run example container image on any dataset that he likes.

### PR DESCRIPTION
**Pull request by gshimansky**
## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #3061  
- [x] tests added and passing
- [x] module layout described at `docs/developer/architecture.rst` is up-to-date

### CODE REVIEW

File: examples/docker/census-on-omnisci/census-omnisci.py [hunk 1]
```diff
@@ -25,25 +26,105 @@
...
+        "DATANUM",
+        "SERIAL",
+        "CBSERIAL",
+        "HHWT",
+        "CPI99",
+        "GQ",
+        "QGQ",
+        "PERNUM",
+        "PERWT",
+        "SEX",
```
Comment (anmyachev) [inline examples/docker/census-on-omnisci/census-omnisci.py:41]: A more compact approach.
```suggestion
        *("YEAR0", "DATANUM", "SERIAL", "CBSERIAL", "HHWT", "CPI99", "GQ", "QGQ", "PERNUM", "PERWT", "SEX"),
```

File: examples/docker/census-on-omnisci/census-omnisci.dockerfile [hunk 2]
```diff
@@ -49,16 +46,12 @@ RUN conda update -n base -c defaults conda -y && \
...
-    conda install -c conda-forge "scikit-learn<0.24.0" && \
+    conda install -c intel/label/modin -c conda-forge -c intel \
+        modin "ray>=1.0.0" "numpy<1.20.0" "scikit-learn<0.24.0" "daal4py>=2021.1" dpcpp_cpp_rt && \
     conda clean --all --yes
 
-COPY ipums_education2income_1970-2010.csv "${HOME}/ipums_education2income_1970-2010.csv"
-
 COPY census-omnisci.py "${HOME}/census-omnisci.py"
+RUN mkdir /dataset
+RUN echo 'conda activate modin && python -u ${HOME}/census-omnisci.py $*' > /entrypoint.sh
```
Comment (anmyachev) [inline examples/docker/census-on-omnisci/census-omnisci.dockerfile:55]: Have you tested passing parameters into `census-omnisci.py`?

File: examples/docker/census-on-omnisci/census-omnisci.dockerfile [hunk 3]
```diff
@@ -49,16 +46,12 @@ RUN conda update -n base -c defaults conda -y && \
...
     conda create -n modin --yes --no-default-packages && \
     conda activate modin && \
-    conda install -c intel/label/modin -c conda-forge modin "ray>=1.0.0" "numpy<1.20.0"
-
-RUN conda activate modin && \
-    conda install -c intel/label/modin -c conda-forge -c intel ${add_extra_channel} \
-        "daal4py>=2021.1" dpcpp_cpp_rt && \
-    conda install -c conda-forge "scikit-learn<0.24.0" && \
+    conda install -c intel/label/modin -c conda-forge -c intel \
+        modin "ray>=1.0.0" "numpy<1.20.0" "scikit-learn<0.24.0" "daal4py>=2021.1" dpcpp_cpp_rt && \
```
Comment (anmyachev) [inline examples/docker/census-on-omnisci/census-omnisci.dockerfile:50]: Creating the environment with two separate commands (`conda install`) was significantly faster than installing with one command. 
How long does it take to create the environment now?

File: examples/docker/census-on-omnisci/census-omnisci.py [hunk 4]
```diff
@@ -143,20 +245,27 @@ def measure(name, func, *args, **kw):
...
     t1 = time.time()
-    print(f'{name}: {t1 - t0} sec')
+    print(f"{name}: {t1 - t0} sec")
     return res
 
 
 def main():
+    if len(sys.argv) != 2:
+        print(
+            f"USAGE: docker run --rm -v /path/to/dataset:/dataset census-omnisci <data file name in /path/to/dataset>"
```
Comment (anmyachev) [inline examples/docker/census-on-omnisci/census-omnisci.py:255]: It seemed to me that in modin it is customary to designate such places filled by the user with `[]` symbols. What do you think?
```suggestion
            f"USAGE: docker run --rm -v /path/to/dataset:/dataset census-omnisci [data file name in /path/to/dataset]"
```

File: examples/docker/plasticc-on-omnisci/plasticc-omnisci.dockerfile [hunk 5]
```diff
@@ -49,21 +46,12 @@ RUN conda update -n base -c defaults conda -y && \
...
-    conda install -c intel/label/modin -c conda-forge "numpy<1.20.0" omniscidbe4py "ray-core>=1.0" \
-        "ray-autoscaler>=1.0" pandas==1.1.5 cloudpickle==1.4.1 rpyc==4.1.5 "dask>=2.12.0" && \
-    conda install -c intel/label/modin -c conda-forge modin==0.8.3
-
-RUN conda activate modin && \
-    conda install -c intel/label/modin -c conda-forge -c intel ${add_extra_channel} \
-        "daal4py>=2021.1" dpcpp_cpp_rt && \
-    conda install -c conda-forge scikit-learn==0.23.2 xgboost && \
+    conda install -c intel/label/modin -c conda-forge -c intel \
+        modin "ray>=1.0.0" "numpy<1.20.0" "scikit-learn<0.24.0" "daal4py>=2021.1" dpcpp_cpp_rt xgboost && \
```
Comment (anmyachev) [inline examples/docker/plasticc-on-omnisci/plasticc-omnisci.dockerfile:50]: OmniSci supports `numpy==1.20.0`. Can you remove the limitation?

**Review by anmyachev — CHANGES_REQUESTED** [review]

File: examples/docker/census-on-omnisci/census-omnisci.dockerfile (hunk 3 above)
Comment (gshimansky) [reply examples/docker/census-on-omnisci/census-omnisci.dockerfile:50]: I tested that 2nd call to conda install is much slower, and 3rd call to conda install finds some conflicts, tries to resolve them and never ends at all. So it is much faster to do in one call. It takes less than 10 minutes to complete after #3052 was merged.

File: examples/docker/census-on-omnisci/census-omnisci.dockerfile (hunk 2 above)
Comment (gshimansky) [reply examples/docker/census-on-omnisci/census-omnisci.dockerfile:55]: Yes I certainly did. This script no longer works when parameters are not passed.

File: examples/docker/census-on-omnisci/census-omnisci.py (hunk 4 above)
Comment (gshimansky) [reply examples/docker/census-on-omnisci/census-omnisci.py:255]: I think square brackets are usually used for optional arguments. This argument is not optional.

File: examples/docker/census-on-omnisci/census-omnisci.py (hunk 1 above)
Comment (gshimansky) [reply examples/docker/census-on-omnisci/census-omnisci.py:41]: It is what black formatting did, I didn't make any changes to the code.

Comment (anmyachev) [reply examples/docker/census-on-omnisci/census-omnisci.py:41]: I know, and yet it is difficult to read code that has a lot of long columns.

File: examples/docker/census-on-omnisci/census-omnisci.dockerfile (hunk 3 above)
Comment (anmyachev) [reply examples/docker/census-on-omnisci/census-omnisci.dockerfile:50]: Good!

File: examples/docker/census-on-omnisci/census-omnisci.py (hunk 1 above)
Comment (gshimansky) [reply examples/docker/census-on-omnisci/census-omnisci.py:41]: Black converts your code into this. Do you really think it is better?
```python
        *(
            "YEAR0",
            "DATANUM",
            "SERIAL",
            "CBSERIAL",
            "HHWT",
            "CPI99",
            "GQ",
            "QGQ",
            "PERNUM",
            "PERWT",
            "SEX",
        ),
```

Comment (anmyachev) [reply examples/docker/census-on-omnisci/census-omnisci.py:41]: In the case tuple should be shorter: `*("YEAR0", "DATANUM", "SERIAL", "CBSERIAL", "HHWT", "CPI99", "GQ", "QGQ")`.

Comment (gshimansky) [reply examples/docker/census-on-omnisci/census-omnisci.py:41]: In my opinion this cosmetic improvement is questionable and is done at an expense of clarity that is important in an example. I don't agree with this change.

File: examples/docker/plasticc-on-omnisci/plasticc-omnisci.dockerfile (hunk 5 above)
Comment (gshimansky) [reply examples/docker/plasticc-on-omnisci/plasticc-omnisci.dockerfile]: Ok I set version to ==1.20.0

**Review by anmyachev — APPROVED** [review]

**Review by devin-petersohn — APPROVED** [review]
LGTM, thanks @gshimansky

**Pull request merged**